### PR TITLE
docs: improve editable table sample

### PR DESCRIPTION
**Pull request by HQFOX**

### CODE REVIEW

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx [hunk 1]
```diff
@@ -432,72 +471,223 @@ const Table = <T extends Data>({
...
+          </HvTableCell>
+          <HvTableCell
+            className={classes.tableCellEditable}
+            key={`editable_row_cell-${cols[2].accessor as string}`}
+            colSpan={1}
+            {...row?.cells[2].getCellProps()}
+          >
+            <div className={edit ? undefined : classes.slide}>
+              <div className={classes.tableCellContent}>
+                <HvDropdown
```
[the remark this answers is not in the document]
Comment (HQFOX) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx:592]: I agree, but maybe we should add something like `inputProps` to keep inline with other form components, what do you think?

[the remark this answers is not in the document]
Comment (MEsteves22) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx:592]: I agree with you 🙏

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx [hunk 2]
```diff
@@ -401,7 +438,9 @@ const Table = <T extends Data>({
       event.preventDefault();
       const formData = new FormData(event.currentTarget);
       const asset = Object.fromEntries(formData.entries());
-      await onRowUpdate?.({ id, ...asset } as T);
+      const formState = editRows.find((r) => r.id === id);
+      const mergedFormData = { ...asset, ...formState };
+      await onRowUpdate?.({ id, ...mergedFormData } as T);
```
[the remark this answers is not in the document]
Comment (HQFOX) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx:441]: I left this because I'm having trouble using the switch. it seems to only register the switch value when it goes from unchecked to checked. So if I were to edit the second row status to `close` it does register the change

[the remark this answers is not in the document]
Comment (MEsteves22) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx:441]: If we remove the `value` property and replace `defaultValue` to:

```tsx
  defaultChecked={edit ? row.original[String(cols[1].accessor)]?.toString() === "Open" : false}
```

I think it helps. Then, on save and edit, we can look at the `status` value to add/edit the row: if it's equal to `on` the status is "Open"; otherwise it's "Closed". What do you think? I'm I missing something?

[the remark this answers is not in the document]
Comment (HQFOX) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx]: Yes, you are wright it works, thanks! I updated it now.

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx [hunk 3]
```diff
@@ -432,72 +469,237 @@ const Table = <T extends Data>({
...
-                      >
-                        <Delete />
-                      </HvButton>
-                    )}
-                  </div>
-                </div>
-              </HvTableCell>
-            );
-          })}
+                <HvSelect
```
[the remark this answers is not in the document]
Comment (HQFOX) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx:639]: I added a enablePortal variable to the `select` component to fix this. It is similar to the variable we have in the `basedropdown` component `disablePortal` but it made sense to me to make this variable off by default as in most cases it doesn't seem to be necessary, what do you think?

File: packages/core/src/Select/Select.tsx [hunk 4]
```diff
@@ -72,6 +73,12 @@ export interface HvSelectProps<
   autoComplete?: string;
   /** Whether the width of the select panel can vary independently. */
   variableWidth?: boolean;
+  /**
+   * Properties passed on to the input element.
+   */
+  inputProps?: React.InputHTMLAttributes<HTMLInputElement>;
```
Comment (zettca) [inline packages/core/src/Select/Select.tsx:79]: I think we should abstract this away from the user. From the users' perspective, there shouldn't be an "underlying input element" (`HvSelect` it the input itself, like HTML's `<select>`).

For now, we could extend from the native element (`Pick<HTMLInputElement, "form">`), but in the future, we should hoist this and any other relevant form props to the `HvFormElement` props

File: packages/core/src/Table/stories/storiesUtils.tsx [hunk 5]
```diff
@@ -60,7 +60,7 @@ const getCell = (value: string) => value as unknown as React.ReactElement;
 
 const formatDate = (date: Date) => date.toISOString().split("T")[0];
 
-const getOption = (opts: string[], i: number) => opts[i % opts.length];
+export const getOption = (opts: string[], i: number) => opts[i % opts.length];
```
Comment (zettca) [inline packages/core/src/Table/stories/storiesUtils.tsx:63]: ?

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx [hunk 6]
```diff
@@ -432,72 +471,228 @@ const Table = <T extends Data>({
...
-                colSpan={last ? 3 : 1}
-                {...row?.cells[idx].getCellProps()}
+          <HvTableCell
+            className={classes.tableCellEditable}
+            key={`editable_row_cell-${cols[0].accessor as string}`}
+            colSpan={1}
+            {...row?.cells[0].getCellProps()}
+          >
+            <div className={edit ? undefined : classes.slide}>
+              <div className={classes.tableCellContent}>
```
Comment (zettca) [inline packages/core/src/Table/stories/TableSection/TableEditable.tsx:481]: We need to review this. The rendering of the cell should be responsibility of the `columns`'s `Cell` property, not hard-coded into the "generic" Table.

If we can't do that, at least this should be refactored into a separate component - it's a lot of code!

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx [hunk 7]
```diff
@@ -595,6 +790,28 @@ const Table = <T extends Data>({
           }}
         />
       ) : undefined}
+      <HvDialog open={openViewDialog} onClose={() => setOpenViewDialog(false)}>
```
Comment (zettca) [inline packages/core/src/Table/stories/TableSection/TableEditable.tsx:793]: This is starting to feel like "too much stuff" for a single sample. This is also not directly related to "editing".

If we can simplify the stuff above, we can keep it - otherwise I think we should move this "View" feature to another sample

File: packages/core/src/Table/stories/TableSection/TableEditable.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (MEsteves22) [reply packages/core/src/Table/stories/TableSection/TableEditable.tsx:639]: It makes also sense to me to keep it off by default 🙏

**Review by zettca — APPROVED** [review]

**Comment by plagoa** [comment]
can we just update this css:
```
  tableRowEditable: css({
    backgroundColor: theme.colors.primary_20,
  }),
```

to:
```
  tableRowEditable: css({
    backgroundColor: theme.colors.containerBackgroundHover,
  }),
```

it's the same value, but that's what makes sense semanticallyty.

**Comment by MEsteves22** [comment]
I noticed that when editing a row, instead of editing it, a new row is added.

**Review by zettca — CHANGES_REQUESTED** [review]

File: packages/core/src/Table/hooks/useTable.ts [hunk 8]
```diff
@@ -506,6 +506,7 @@ export interface HvColumnInstance<
     Partial<UseResizeColumnsColumnProps<D>>,
     Partial<UseSortByColumnProps<D>> {
   Cell?: Renderer<HvCellProps<D, H>>;
+  accessor: HvColumnInstance<D, H>;
```
[the remark this answers is not in the document]
Comment (HQFOX) [reply packages/core/src/Table/hooks/useTable.ts:509]: I was having trouble with the column type of the `EditableCell` , when it is rendering the row in "add" mode it has that property but it is not in the type.

**Review by zettca — APPROVED** [review]

**Review by MEsteves22 — APPROVED** [review]
👌🏻
